pyproject.toml

```diff
@@ -18,6 +18,18 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[tool.mypy]
+files = "trieste,tests,notebooks/**/*.py"
+ignore_missing_imports = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+strict_equality = true
+
+[[tool.mypy.overrides]]
+module = ["trieste.*", "tests.*"]
+disallow_any_generics = true
+disallow_untyped_defs = true
```

Comment:
these will mean contributors can't merge if they haven't fully specified the types

Reply:
I support this, it does increase the barrier on contributors, but it doesn't make sense we do typing on users' PR's